### PR TITLE
:sparkles: bump controller-runtime to 0.8.3

### PR DESCRIPTION
**Pull request by varshaprasad96**
deps: Bump controller-runtime to 0.8.0

### CODE REVIEW

**Comment by varshaprasad96** [comment]
Blocked on #[146](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/146). Need to regenerate test data for `project-v3-addon` and `project-v3-config` plugins.

File: testdata/project-v3-addon/PROJECT
Comment (camilamacedo86) [inline testdata/project-v3-addon/PROJECT:20]: Hi @varshaprasad96, 

Shows that have something broked with the addon partner. we might to need to do changes first in sigs.k8s.io/kubebuilder-declarative-pattern.

c/c @estroz

Comment (varshaprasad96) [reply testdata/project-v3-addon/PROJECT:20]: Yes, the samples for v3-addon need to be regenerated after https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/146 is merged

**Comment by Adirio** [comment]
Note: current PR title and description checks do not understand the `[WIP]` part, so they will fail anyway, but once you remove it, the emoji (✨ `:sparkles:`)  is expected without brackets, so you will also have to remove those.

P.S.: the next version of the PR title verifier will understand the [WIP] prefix.

**Comment by varshaprasad96** [comment]
The PR has been updated. Waiting for v3.0.0 release.
/hold

**Comment by Adirio** [comment]
@varshaprasad96 Could you remove the brackets from the title's emoji? Either ✨ or :sparkles: is fine, (and GitHub started understanding emojis on titles).

**Comment by camilamacedo86** [comment]
HI @estroz, 

You spoke about we push it in 3.1.0. However, wdyt? Could it get merged now or should we wait? if it gets merged than it means that if we need to rebase master with the release-v3 before 3.0.0 it will ship as well. I have not an objection at all with 3.0.0 be using controller-runtime 0.8.0.

**Comment by Adirio** [comment]
The reason that was given (I think it was @estroz but not 100% sure) was that there is no kubebuilder version that supports v0.7.0 which is the first version supporting kubernetes 1.19.x so that it would be nicer to have a v3.0.0 with v0.7.0 and then upgrade it for v3.1.0.

**Review by Adirio — APPROVED** [review]
/ok-to-test

**Review by camilamacedo86 — DISMISSED** [review]
I am ok with it get merged. I do not think that would be a requirement we have any version of the tool using 0.7.0. 

@varshaprasad96 could you only please update the migration doc and bump the version 0.8.0 where we have 0.7.0? 

c/c @estroz

**Comment by varshaprasad96** [comment]
Based on the discussion, I am reverting the migration doc changes, since the default version for v3 stable would be c-r 0.7.0

**Comment by estroz** [comment]
>since the default version for v3 stable would be c-r 0.7.0

@varshaprasad96 yes but after this PR merges it will be 0.8.0 right?

**Comment by Adirio** [comment]
Even if the docs are changed in master they won't get reflected in the published book until a new release that contains this merged PR is made, and then the version would be 0.8.0.

**Comment by camilamacedo86** [comment]
@varshaprasad96 @estroz,

This PR bumps the controller-runtime  to 0.8.0
When it gets merged we need also update the docs
The production docs are only updated as part of the release where we push the master changes to the book branch. So, the master docs need to reflect the master state and update the doc is part of the PR. If we need to revert we will also revert the docs changes. 

Is it make sense?

**Comment by varshaprasad96** [comment]
Sorry, got confused ! Updated the migration docs

**Comment by camilamacedo86** [comment]
We need also have the commits squashed as well.

**Comment by estroz** [comment]
@varshaprasad96 this can be rebased and bumped to controller-runtime v0.8.3

**Comment by varshaprasad96** [comment]
@estroz Done. bumped to controller-runtime v0.8.3

**Pull request merged**
